pkg/service/oidc_provider_test.go

```diff
@@ -0,0 +1,254 @@
+	uuuid := uuid.New().String()
+	uuuuid := uuid.New().String()
+
+	mock.ExpectQuery(`SELECT "oidcprovider"."id", "oidcprovider"."name"`).
+		WithArgs().WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(uuuid))
+
+	provider := &systemv3.OIDCProvider{
+		Metadata: &v3.Metadata{Id: uuuid, Name: "user-" + uuuuid},
+		Spec:     &systemv3.OIDCProviderSpec{ProviderName: "provider-" + pruuid},
+	}
+
+	_, err := OP.GetByName(context.Background(), provider)
+	if err != nil {
+		t.Fatal("could not get partner:", err)
+	}
+	performOidcProviderBasicChecks(t, provider, uuuuid, pruuid)
+}
+
+func TestOidcProviderUpdate(t *testing.T) {
+	db, mock := getDB(t)
+	defer db.Close()
+
+	OP := NewOIDCProviderService(db, "", getLogger())
+
+	uuuid := uuid.New().String()
+	uuuuid := uuid.New().String()
+	pruuid := uuid.New().String()
+
+	scope := []string{"email"}
+
+	puuid, ouuid := addParterOrgFetchExpectation(mock)
+
+	mock.ExpectQuery(`SELECT "oidcprovider"."id", "oidcprovider"."name", .* FROM "authsrv_oidc_provider" AS "oidcprovider" WHERE`).
+		WithArgs().WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(uuuuid))
+
+	mock.ExpectExec(`UPDATE "authsrv_oidc_provider" AS "oidcprovider" SET "name" = 'user-` + uuuid + `', .*"organization_id" = '` + ouuid + `', "partner_id" = '` + puuid + `.* WHERE \(id = '` + uuuuid + `'\)`).
+		WillReturnResult(sqlmock.NewResult(1, 1))
+
+	provider := &systemv3.OIDCProvider{
+		Metadata: &v3.Metadata{Id: uuuuid, Name: "user-" + uuuid, Partner: "partner-" + puuid, Organization: "org-" + ouuid},
+		Spec:     &systemv3.OIDCProviderSpec{Scopes: scope, IssuerUrl: "https://token.actions.githubusercontent.com", ProviderName: "provider-" + pruuid},
+	}
+
+	_, err := OP.Update(context.Background(), provider)
+	if err != nil {
+		t.Fatal("could not update provider:", err)
+	}
+	performOidcProviderBasicChecks(t, provider, uuuid, pruuid)
+}
+
+func TestOidcProviderUpdateInvalidUrl(t *testing.T) {
+	db, mock := getDB(t)
+	defer db.Close()
+
+	OP := NewOIDCProviderService(db, "", getLogger())
+
+	uuuid := uuid.New().String()
+	uuuuid := uuid.New().String()
+	pruuid := uuid.New().String()
+
+	scope := []string{"email"}
+
+	puuid, ouuid := addParterOrgFetchExpectation(mock)
+
+	mock.ExpectQuery(`SELECT "oidcprovider"."id", "oidcprovider"."name", .* FROM "authsrv_oidc_provider" AS "oidcprovider" WHERE`).
+		WithArgs().WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(uuuuid))
+
+	mock.ExpectExec(`UPDATE "authsrv_oidc_provider" AS "oidcprovider" SET "name" = 'user-` + uuuid + `', .*"organization_id" = '` + ouuid + `', "partner_id" = '` + puuid + `.* WHERE \(id = '` + uuuuid + `'\)`).
+		WillReturnResult(sqlmock.NewResult(1, 1))
+
+	provider := &systemv3.OIDCProvider{
+		Metadata: &v3.Metadata{Id: uuuuid, Name: "user-" + uuuid, Partner: "partner-" + puuid, Organization: "org-" + ouuid},
+		Spec:     &systemv3.OIDCProviderSpec{Scopes: scope, IssuerUrl: "test.url", ProviderName: "provider-" + pruuid, MapperUrl: "test.url"},
```

Comment:
I think we could probably create a [table driven test](https://mj-go.in/golang/table-driven-tests-in-go) for the error cases to test them individually.